# Paper_figures

To create the following images, it was essential to use epigenetic (ChIP-seq) and transcriptomic (RNA-seq) data from Kuang *et al.* (2014), 
chromatin accessibility information from ATAC-seq (Gowans *et al.*, 2018), and the *S. cerevisiae* metabolic network information provided by 
the BiGG Models repository (King *et al.*, 2015).

*Figure S1 was adapted from the methodology used by Sánchez *et al.* (2018). 


# References

Gowans, G. J., Schep, A. N., Wong, K. M., King, D. A., Greenleaf, W. J., & Morrison, A. J. (2018). INO80 chromatin remodeling coordinates metabolic homeostasis with cell division. Cell reports, 22(3), 611-623.

King, Z. A., Lu, J. S., Dräger, A., Miller, P., Federowicz, S., Lerman, J. A., Ebrahim, A., Palsson, B. Ø., & Lewis, N. E. (2015). BIGG Models: a platform for integrating, standardizing and sharing genome-scale models. *Nucleic Acids Research*, 44(D1), D515-D522. https://doi.org/10.1093/nar/gkv1049.

Kuang, Z., Cai, L., Zhang, X., Ji, H., Tu, B.P. and Boeke, J.D., 2014. High-temporal-resolution view of transcription and chromatin states across distinct metabolic states in budding yeast. *Nature structural & molecular biology*, 21(10), pp.854-863.

Sánchez-Gaya, V., Casaní-Galdón, S., Ugidos, M., Kuang, Z., Mellor, J., Conesa, A., & Tarazona, S. (2018). Elucidating the role of chromatin state and transcription factors on the regulation of the yeast metabolic Cycle: a Multi-OmiC Integrative Approach. *Frontiers in Genetics*, 9. https://doi.org/10.3389/fgene.2018.00578
